src/main/java/net/glowstone/util/lang/I.java

```diff
@@ -0,0 +1,176 @@
+        } catch (MissingResourceException ex) {
+            throw new MissingResourceException("Missing locale '" + locale + "'. class=" + ex.getClassName(), ex.getClassName(), ex.getKey());
+        }
+        defaultLocale = locale;
+    }
+
+    /**
+     * Returns the internationalized message according to the effective locale of the player.
+     *
+     * @param sender CommandSender used when looking up locale.
+     * @param key Language translation key mapping to a specific string in the translations.
+     * @param args Arguments used by the formatter to replace placeholder variables. If there are more arguments than format specifiers, the extra arguments are ignored.
+     */
+    public static String tr(CommandSender sender, String key, Object ... args) {
+        return trl(getEffectiveLocale(sender), key, args);
+    }
+
+    /**
+    * Returns the internationalized message according to the default locale of the server.
+     *
+     * @param key Language translation key mapping to a specific string in the translations.
+     * @param args Arguments used by the formatter to replace placeholder variables. If there are more arguments than format specifiers, the extra arguments are ignored.
+    */
+    public static String tr(String key, Object ... args) {
+        return trl(defaultLocale, key, args);
+    }
+
+    /**
+    * Returns the internationalized message according to the specified locale.
+     *
+     * @param locale Locale to use when looking up translation.
+     * @param key Language translation key mapping to a specific string in the translations.
+     * @param args Arguments used by the formatter to replace placeholder variables. If there are more arguments than format specifiers, the extra arguments are ignored.
+     * @throws MissingResourceException Thrown if the provided specified does not exist.
+    */
+    public static String trl(@NonNull String locale, @NonNull String key, @NonNull Object ... args) {
+        ResourceBundle rb;
+        try {
+            rb = ResourceBundle.getBundle(BASE_NAME, Locale.forLanguageTag(locale));
+        } catch (MissingResourceException ex) {
+            throw new MissingResourceException("Missing locale '" + locale + "'. class=" + ex.getClassName(), ex.getClassName(), ex.getKey());
+        }
+
+        String result;
+        if (rb.containsKey(key)) {
+            result = rb.getString(key);
+        } else {
+            ResourceBundle fallback = ResourceBundle.getBundle(BASE_NAME, Locale.forLanguageTag(FALLBACK_LOCALE));
+            if (fallback.containsKey(key)) {
+                result = fallback.getString(key);
+            } else {
+                throw new MissingTranslationException("Could not find translation for key '" + key + "' in fallback locale", locale, key);
+            }
+        }
+        for (int i = 0; i != args.length; i++) {
+            result = result.replaceAll("\\{" + i + "\\}", args[i].toString());
+        }
+        return result;
```

Comment:
For each call you need to create a Pair<key:String,color:ChatColor>. Check if that worked. Then remove each element where ^ChatColor\.[A-Z]* $   It will be some work.